Packs/OpenAI/Integrations/OpenAI/OpenAI.py

```diff
@@ -123,41 +299,184 @@ def reputations_command(client: Client, args: dict) -> CommandResults:
+    answer = result['answer']
+    return CommandResults(
+        outputs_prefix='OpenAI.QA',
+        outputs={
+            "Question": question,
+            "Answer": answer
+        },
+        readable_output=f"**Answer**: {answer}"
+    )
+
+
+def summarize_command(client: Client, args: dict) -> str:
+    """
+        Summarize the provided input data using LangChain:
+        https://python.langchain.com/docs/modules/chains/popular/summarize.html
+
+        Args:
+            client (Client): OpenAI client object
+            args (dict): Arguments passed into the command
+
+        Returns:
+            str: Summary generated by the LLM
+    """
+    text = args.get("text", "")
+    entry_id = args.get("entry_id", "")
+    prompt = args.get("prompt", "")
+    model = args.get("model")
+    deployment = args.get("deployment")
+    temperature = args.get("temperature")
+
+    # load in the documents to summarize
+    docs = process_text_input_args(text, entry_id)
+    if client.is_azure:
+        llm = AzureOpenAI(
+            temperature=temperature,
+            deployment_name=deployment,
+            model_name=model
+        )
+    else:
+        llm = OpenAI(temperature=temperature)
+    # user supplied an additional prompt with specific instructions for how to summarize
+    if prompt:
+        if "{text}" not in prompt:
+            raise DemistoException("Prompt must include the string {text} to indicate the position of the text input data within the prompt.")
+        prompt = PromptTemplate(template=prompt, input_variables=["text"])
+        chain = load_summarize_chain(llm=llm, chain_type="map_reduce",
+                                     map_prompt=prompt, combine_prompt=prompt)
+    else:
+        chain = load_summarize_chain(llm=llm, chain_type="map_reduce")
+    result = chain.run(docs).lstrip()  # remove leading whitespace from the result
+    return CommandResults(
+        outputs_prefix='OpenAI.Summary',
+        outputs=result,
+        readable_output=f"**Summary**: {result}"
+    )
+
+
 ''' MAIN FUNCTION '''
 
 
 def main() -> None:
-    """main function, parses params and runs command functions
-
+    """main function, runs command functions
     """
     params = demisto.params()
     args = demisto.args()
     command = demisto.command()
 
-    base_url = params.get('url')
     api_key = params.get('apikey')
+    base_url = params.get('url', '')
+    if base_url[-1] != '/':
+        base_url = base_url + '/'
+    is_azure = params.get('is_azure', False)
     verify = not params.get('insecure', False)
     proxy = params.get('proxy', False)
+    version = params.get('version', '2023-03-15-preview')
+    test_model = params.get('test_model', '')
 
-    try:
-        client = Client(
-            base_url=base_url,
-            api_key=api_key,
-            verify=verify,
-            proxy=proxy
-        )
+    # set env variable for API key
+    os.environ["OPENAI_API_KEY"] = api_key
```

Comment:
What is the reason of setting env variables? I saw that for the [FAISS](https://python.langchain.com/docs/modules/data_connection/vectorstores/integrations/faiss) library, this is needed, are the other env variables for the Azure OpenAI vendor?

Reply:
Yes exactly, for FAISS and for using Azure OpenAI with LangChain: https://python.langchain.com/docs/integrations/providers/azure_openai